src/GovUk.Education.ExploreEducationStatistics.Content.Services/MethodologyService.cs

```diff
@@ -125,5 +131,82 @@ private async Task<List<MethodologyVersionSummaryViewModel>> BuildMethodologiesF
                 await _methodologyVersionRepository.GetLatestPublishedVersionByPublication(publicationId);
             return _mapper.Map<List<MethodologyVersionSummaryViewModel>>(latestPublishedMethodologies);
         }
+
+        // This method is responsible for keeping Methodology Titles and Slugs in sync with their owning Publications
+        // where appropriate.  Methodologies always keep track of their owning Publication's titles and slugs for
+        // optimisation purposes.
+        public async Task PublicationTitleOrSlugChanged(Guid publicationId, string originalSlug, string updatedTitle,
+            string updatedSlug)
+        {
+            var slugChanged = originalSlug != updatedSlug;
+
+            var ownedMethodology = await _contentDbContext
+                .PublicationMethodologies
+                .Include(pm => pm.Methodology.LatestPublishedVersion)
+                .Include(pm => pm.Methodology.Versions)
+                .Where(pm => pm.PublicationId == publicationId && pm.Owner)
+                .Select(pm => pm.Methodology)
+                .SingleOrDefaultAsync();
+
+            if (ownedMethodology == null)
+            {
+                return;
+            }
+
+            ownedMethodology.OwningPublicationTitle = updatedTitle;
+            ownedMethodology.OwningPublicationSlug = updatedSlug;
+
+            _contentDbContext.Methodologies.Update(ownedMethodology);
+
+            if (slugChanged)
+            {
+                // A redirect is only needed for the LatestPublishedVersion.
+                // Unpublished methodology's don't need a redirect - they're not live.
+                // An unpublished amendment doesn't need a redirect because:
+                // - if it uses OwningPublicationSlug, it is covered by the LatestPublishedVersion redirect created here
+                // - if it uses AlternativeSlug, a redirect would have been created at the time the AlternativeSlug
+                //   was set (and that redirect will become active when that version is published).
+                if (ownedMethodology.LatestPublishedVersion is { AlternativeSlug: null }
```

Comment:
Nice pattern matching here :)  I should use it more!

Reply:
I only used it because Rider suggested it, to be honest :smile: